lib/async_hooks.js

```diff
@@ -476,6 +476,31 @@ function init(asyncId, type, triggerAsyncId, resource) {
   processing_hook = false;
 }
 
+// Integration with V8 Inspector AsyncTask* API
+
+const inspectorHook = createHook({
+  init(asyncId, type, triggerId, resource) {
+    // TODO(bajtos): support recurring tasks like setInterval
+    const recurring = false;
+    async_wrap.asyncTaskScheduled(type, asyncId, recurring);
+  },
+
+  before(asyncId) {
+    async_wrap.asyncTaskStarted(asyncId);
+  },
+
+  after(asyncId) {
+    async_wrap.asyncTaskFinished(asyncId);
+  },
+
+  destroy(asyncId) {
+    async_wrap.asyncTaskCanceled(asyncId);
+  },
+});
+
+// TODO(bajtos): enable the hook when Inspector is started,
+// disable it when Inspector is stopped
+inspectorHook.enable();
```

Comment:
I definitely like the idea here but we should definitely ask the question: should this be turned on by default or on demand only? Should there be an API for disabling it if it is on by default?

I'd vote for on by default, no api for disabling.